server/middleware/namespace.go

```diff
@@ -18,55 +18,54 @@ import (
 	"context"
 
 	middleware "github.com/grpc-ecosystem/go-grpc-middleware"
-	api "github.com/tigrisdata/tigris/api/server/v1"
 	"github.com/tigrisdata/tigris/lib/container"
-	"github.com/tigrisdata/tigris/server/config"
-	"github.com/tigrisdata/tigris/server/metadata"
 	"github.com/tigrisdata/tigris/server/request"
 	"google.golang.org/grpc"
 )
 
-type NamespaceSetter struct {
-	tenantManager      *metadata.TenantManager
-	namespaceExtractor request.NamespaceExtractor
-	excludedMethods    container.HashSet
-	config             *config.Config
-}
+var (
+	excludedMethods = container.NewHashSet(
+		"/HealthAPI/Health",
+		"/tigrisdata.admin.v1.Admin/createNamespace",
+		"/tigrisdata.admin.v1.Admin/listNamespaces",
+	)
+
+	namespaceExtractor = &request.AccessTokenNamespaceExtractor{}
+)
 
-func (r *NamespaceSetter) NamespaceSetterUnaryServerInterceptor() func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+func namespaceSetterUnaryServerInterceptor(enabled bool) func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-		if !r.config.Auth.EnableNamespaceIsolation || r.excludedMethods.Contains(info.FullMethod) {
-			return handler(request.SetNamespace(ctx, request.DefaultNamespaceName), req)
-		} else {
-			namespace, err := r.namespaceExtractor.Extract(ctx)
-			if err != nil {
-				return nil, err
-			}
-			if namespace == "" {
-				return handler(request.SetNamespace(ctx, "unknown"), req)
+		namespace := request.DefaultNamespaceName
+		if enabled && !excludedMethods.Contains(info.FullMethod) {
+			var err error
+			if namespace, err = namespaceExtractor.Extract(ctx); err != nil {
+				// We know that getAccessToken with app_id/app_secret credentials doesn't have namespace set.
+				// Mark it as default_namespace instead of unknown.
+				if info.FullMethod != "/tigrisdata.auth.v1.Auth/getAccessToken" {
+					// We return and error when the token is set, but namespace is empty
+					return nil, err
+				}
+				namespace = request.DefaultNamespaceName
 			}
-			return handler(request.SetNamespace(ctx, namespace), req)
 		}
+
+		return handler(request.SetNamespace(ctx, namespace), req)
```

Comment:
I guess this is just for metric reporting?

Reply:
This is for everything where namespace is needed downstream.